taichi/codegen/spirv/spirv_ir_builder.cpp

```diff
@@ -415,13 +423,20 @@ SType IRBuilder::get_sampled_image_type(const SType &primitive_type,
                /*Depth=*/0, /*Arrayed=*/0, /*MS=*/0, /*Sampled=*/1,
                spv::ImageFormatUnknown)
       .commit(&global_);
+
   SType sampled_t;
   sampled_t.id = id_counter_++;
   sampled_t.flag = TypeKind::kImage;
   ib_.begin(spv::OpTypeSampledImage)
       .add_seq(sampled_t, img_id)
```

Comment:
Then here, we can get the image type value by calling `get_underlying_image_type`.

In a IR builder API, when we call a function `get_xxx_type` we expect to get a value that holds the type, and this value can be cached from a previous construction. But the gist is calling this function shouldn't fail and should always get us the correct type.